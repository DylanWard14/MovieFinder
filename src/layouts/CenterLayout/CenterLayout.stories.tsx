import React from 'react';

import CenterLayout from './CenterLayout';

export default {
    title: 'Layouts/CenterLayout',
};

export const DefaultExample = () => (
    <CenterLayout>
        <div>
            <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc pretium interdum ex ut pulvinar. Ut semper
                turpis tortor, porta euismod diam mattis ac. Etiam finibus libero vel tortor feugiat egestas. Morbi
                aliquet dictum condimentum. Suspendisse lobortis tempor ligula vitae imperdiet.
            </p>
        </div>
    </CenterLayout>
);
