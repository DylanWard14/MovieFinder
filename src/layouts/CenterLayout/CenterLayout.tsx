import React from 'react';
import styled from 'styled-components/macro';

interface CenterLayoutProps {
    children: React.ReactNode;
}

const Component = styled.div`
    width: 50vw;
    margin: 0 auto;
`;

const CenterLayout: React.FunctionComponent<CenterLayoutProps> = ({children}) => {
    return (
        <Component>
            {children}
        </Component>
    );
}

export default CenterLayout;