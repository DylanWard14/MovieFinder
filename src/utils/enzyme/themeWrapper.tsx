import React from 'react';
import { render, shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import myTheme from '../../theme';

export const renderWrapper = (component: React.ReactNode) => {
    return render(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const shallowWrapper = (component: React.ReactNode) => {
    return shallow(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const mountWrapper = (component: React.ReactNode) => {
    return mount(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};
