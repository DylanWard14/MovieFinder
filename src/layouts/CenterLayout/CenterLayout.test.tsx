import React from 'react';
import { render } from 'enzyme';

import CenterLayout from './CenterLayout';

describe('CenterLayout', () => {
    it('renders default correctly', () => {
        const wrapper = render(
            <CenterLayout>
                <div>
                    <p>
                        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc pretium interdum ex ut pulvinar.
                        Ut semper turpis tortor, porta euismod diam mattis ac. Etiam finibus libero vel tortor feugiat
                        egestas. Morbi aliquet dictum condimentum. Suspendisse lobortis tempor ligula vitae imperdiet.
                    </p>
                </div>
            </CenterLayout>,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
