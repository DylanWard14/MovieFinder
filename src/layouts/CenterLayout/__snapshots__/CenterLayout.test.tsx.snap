// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenterLayout renders default correctly 1`] = `
<div
  class="CenterLayout__Component-sc-1uihjrr-0 iyaool"
>
  <div>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc pretium interdum ex ut pulvinar. Ut semper turpis tortor, porta euismod diam mattis ac. Etiam finibus libero vel tortor feugiat egestas. Morbi aliquet dictum condimentum. Suspendisse lobortis tempor ligula vitae imperdiet.
    </p>
  </div>
</div>
`;
